### ISSUE TITLE
Verification of KDC fails when using SHA384 digest signed certificates

### ISSUE THREAD

**Issue by mcleoddr** [heimdal/heimdal#776]
We recently upgraded our KDC lab (Windows) to use certificates signed with SHA384WithRSAEncryption. Our client using Heimdal 7.7 started returning HX509_SIG_ALG_NO_SUPPORTED during TGT fetch for PKINIT.

The API that returned HX509_SIG_ALG_NO_SUPPORTED is krb5_init_creds_get().

The bug is in heimdal/lib/hx509/crypto.c:

Actual:
```
static const struct signature_alg sha384_alg = {
    "sha-384",    
    ASN1_OID_ID_SHA512,   
    &amp;_hx509_signature_sha384_data,    
    NULL,    
    NULL,    
    SIG_DIGEST,    
    0,
    EVP_sha384,    
    evp_md_verify_signature,    
    evp_md_create_signature,
    0};
```

Expected:
```
static const struct signature_alg sha384_alg = {
    "sha-384",    
    ASN1_OID_ID_SHA384,   
    &amp;_hx509_signature_sha384_data,    
    NULL,    
    NULL,    
    SIG_DIGEST,    
    0,
    EVP_sha384,    
    evp_md_verify_signature,    
    evp_md_create_signature,
    0};
```

That is, the second member (sig_oid) of sha384_alg struct should be ASN1_OID_ID_SHA**384**, not ASN1_OID_ID_SHA512.

When we made this fix on our client we could successfully obtain a TGT.

**Comment by lhoward** [comment]
Looks like a typo in the code to me, @nicowilliams?

**Comment by jaltman** [comment]
@lhoward it certainly looks like a copy/paste bug that was introduced during the original commit of sha384 3128a7a41653bd3ee99bfd289622737cbdb566ee

As far as I can see it has not been fixed by Apple.

**Comment by lhoward** [comment]
Fixed in 718e3f8b6883c8332b55f864d057ca80b2541ded (forgot the magic pound symbol so not auto-closed).

**Comment by lhoward** [comment]
BTW should we add `id-pkinit-kdf-ah-sha384` to `pkinit.asn1` and `lib/krb5/crypto-pk.c`?
